src/cargo/ops/fix.rs

```diff
@@ -321,17 +322,31 @@ fn check_resolver_change(ws: &Workspace<'_>, opts: &FixOptions) -> CargoResult<(
 }
 
 fn report_maybe_diesel(config: &Config, resolve: &Resolve) -> CargoResult<()> {
-    if resolve
-        .iter()
-        .any(|pid| pid.name() == "diesel" && pid.version().major == 1)
-        && resolve.iter().any(|pid| pid.name() == "diesel_migrations")
-    {
+    fn is_broken_diesel(pid: PackageId) -> bool {
+        if pid.name() != "diesel" {
+            return false;
+        }
+        Comparator {
+            op: Op::Less,
+            major: 1,
+            minor: Some(4),
+            patch: Some(8),
+            pre: Prerelease::EMPTY,
+        }
+        .matches(pid.version())
```

Comment:
I think this can be simplified a bit by just comparing with a Version:

```suggestion
        pid.name() == "diesel" && pid.version() < &Version::new(1, 4, 8)
```

Reply:
Done :+1: